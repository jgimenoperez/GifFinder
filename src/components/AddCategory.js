import React, {useState} from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ( {setcategories} ) => {

    const [inputvalue, setinputvalue] = useState(``)

    const handleImputChange=(e)=>{

            console.log('handleImputChange llamado')
            setinputvalue(e.target.value)
    }

    const hasdleSubmit=(e)=>{

            e.preventDefault()
            console.log('Handel submit')

            if (inputvalue.trim().length>2)
            {
                setcategories( cats => [inputvalue,...cats,]);      
                setinputvalue('')
            }
            // console.log('Submit hecho')
            
            // console.log(props.otracosa)
    }



    return (
        <div>

           <form onSubmit={hasdleSubmit}>

                <p>{inputvalue}</p>

                <input
                    type="text"
                    value={inputvalue}
                    onChange={ handleImputChange }
                />

            </form>    

            
        </div>
    )
}
 
AddCategory.propTypes = {

    setcategories:PropTypes.func.isRequired

}
