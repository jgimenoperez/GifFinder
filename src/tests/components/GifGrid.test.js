import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid"
import { useFetchgifs } from '../../hooks/useFetchgifs';
jest.mock('../../hooks/useFetchgifs');


describe('Pruebas en el Gifgrid', () => {

    const categorie = "One Punch"
    
    test('Debe de mostarse correctamente ', () => {


        useFetchgifs.mockReturnValue({

            data:[],
            loading: true
        })


        const wrapper = shallow (<GifGrid categorie={categorie}/>)
        expect( wrapper ).toMatchSnapshot()

        
    })


    test('debe de mostrar items cuando se cargan imagenes userFetch', () => {

        const gifs =[{

            id: '123',
            url: 'https:\\www.manosdehada.es ' ,
            title: 'lo que sea'

        }]    


        useFetchgifs.mockReturnValue({

            data:gifs,
            loading: false
        })
        
        const wrapper = shallow (<GifGrid categorie={categorie}/>)

        expect ( wrapper ).toMatchSnapshot();
        expect( wrapper.find('P').exists() ).toBe(false)
        expect( wrapper.find('GifGridItem').length).toBe(gifs.length)
        
    })
    
    

})
