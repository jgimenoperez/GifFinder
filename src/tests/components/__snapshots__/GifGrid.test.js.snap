// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Gifgrid Debe de mostarse correctamente  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__shakeY"
  >
     
    One Punch
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en el Gifgrid debe de mostrar items cuando se cargan imagenes userFetch 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__shakeY"
  >
     
    One Punch
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="123"
      key="123"
      title="lo que sea"
      url="https:\\\\www.manosdehada.es "
    />
  </div>
</Fragment>
`;
